test/plugins/sqlite.js

```diff
@@ -0,0 +1,102 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const expect = require("chai").expect;
+const Msg = require("../../src/models/msg");
+const Helper = require("../../src/helper");
+const MessageStorage = require("../../src/plugins/sqlite.js");
+
+describe("SQLite Message Storage", function() {
+	const expectedPath = path.join(Helper.getHomePath(), "logs", "testUser.sqlite3");
+	let store;
+
+	// Delete database file from previous test run
+	before((done) => {
+		store = new MessageStorage();
+
+		if (fs.existsSync(expectedPath)) {
+			fs.unlink(expectedPath, done);
+		} else {
+			done();
+		}
+	});
+
+	it("should resolve an empty array when disabled", (done) => {
+		store.getMessages(null, null).then((messages) => {
+			expect(messages).to.be.empty;
+			done();
+		});
+	});
+
+	it("should create database file", () => {
+		expect(store.isEnabled).to.be.false;
+		expect(fs.existsSync(expectedPath)).to.be.false;
+
+		store.enable("testUser");
+
+		expect(store.isEnabled).to.be.true;
+		expect(fs.existsSync(expectedPath)).to.be.true;
+	});
+
+	it("should create tables", (done) => {
+		store.database.serialize(() =>
+			store.database.all("SELECT name, tbl_name, sql FROM sqlite_master WHERE type = 'table'", (err, row) => {
+				expect(err).to.be.null;
+				expect(row).to.deep.equal([{
+					name: "options",
+					tbl_name: "options",
+					sql: "CREATE TABLE options (name TEXT, value TEXT, CONSTRAINT name_unique UNIQUE (name))",
+				},
+				{
+					name: "messages",
+					tbl_name: "messages",
+					sql: "CREATE TABLE messages (network TEXT, channel TEXT, time INTEGER, type TEXT, msg TEXT)",
+				}]);
+
+				done();
+			})
+		);
+	});
+
+	it("should insert schema version to options table", (done) => {
+		store.database.serialize(() =>
+			store.database.get("SELECT value FROM options WHERE name = 'schema_version'", (err, row) => {
+				expect(err).to.be.null;
+
+				// Should be sqlite.currentSchemaVersion,
+				// compared as string because it's returned as such from the database
+				expect(row.value).to.equal("1");
+
+				done();
+			})
+		);
+	});
+
+	it("should store a message", (done) => {
+		store.index("this-is-a-network-guid", "#ThisIsAChannel", new Msg({
+			time: 123456789,
+			text: "Hello from sqlite world!",
+		}));
+
+		store.database.serialize(done);
+	});
+
+	it("should retrieve previously stored message", (done) => {
+		store.getMessages({
```

Comment:
This retrieves a message stored in a previous test?!
It would be better if tests didn't depend on each other. You'll also need to change `before` into `beforeEach` above.

Reply:
I really don't see a need to do that (it would be more work than just changing `before`). I understand it's great when each test is self contained but that's not always practical.

Reply:
tbh if it's not practical, there might be a problem somewhere. In my experience, this _will_ bite us.
I understand it's extra work though, so I'm not gonna veto or anything, but I'd be more comfortable if we don't do that (I can lend a hand if you want). @McInkay, what's your take on this?